server/cluster_worker_test.go

```diff
@@ -308,6 +308,21 @@ func mustGetRegion(c *C, cluster *raftCluster, key []byte, expect *metapb.Region
 	c.Assert(r, DeepEquals, expect)
 }
 
+func checkSearchRegions(c *C, cluster *raftCluster, keys ...[]byte) {
+	cacheRegions := cluster.cachedCluster.regions
+	c.Assert(cacheRegions.searchRegions.Len(), Equals, len(keys))
+
+	for _, key := range keys {
+		mockRegion := &metapb.Region{EndKey: key}
+		item := &searchKeyItem{
+			key: searchKey(encodeRegionEndKey(mockRegion)),
+		}
+
+		getItem := cacheRegions.searchRegions.Get(item)
+		c.Assert(getItem.(*searchKeyItem).key, DeepEquals, item.key)
+	}
```

Comment:
I think it's better to iterate `searchRegions` and compare each item with the expect keys, then we can be sure there is no extra items in the btree.